docker/release/config/opensearch/opensearch-docker-entrypoint.sh

```diff
@@ -34,8 +34,10 @@ function setupSecurityPlugin {
         if [ "$DISABLE_INSTALL_DEMO_CONFIG" = "true" ]; then
             echo "Disabling execution of install_demo_configuration.sh for OpenSearch Security Plugin"
         else
-            echo "Enabling execution of install_demo_configuration.sh for OpenSearch Security Plugin"
-            bash $OPENSEARCH_HOME/plugins/$SECURITY_PLUGIN/tools/install_demo_configuration.sh -y -i -s
+            echo "OpenSearch 2.12.0 onwards, the OpenSearch Security Plugin a change that requires an initial password for 'admin' user."
+            echo "Please define an environment variable 'OPENSEARCH_INITIAL_ADMIN_PASSWORD' with a strong password string."
+            echo "If a password is not provided, the setup will quit."
+            bash $OPENSEARCH_HOME/plugins/$SECURITY_PLUGIN/tools/install_demo_configuration.sh -y -i -s || exit 1
```

Comment:
Similar to #4250.

Reply:
based on offline discussion we are going to avoid adding `set -e` to exit the setup scripts, instead we will use `|| exit 1` to hard stop the script in case of any failures with demo script

Reply:
`echo -e` should be more posix compliant than `printf`.